### ISSUE TITLE
Do not change order of columns in schema file based on DB column order

### ISSUE THREAD

**Issue by SharadKumar** [prisma/prisma#3623]
## Problem
With `v2.6.2`, I noticed that when you introspect... my schema (inc relations/name) maintained. However, the ordering of columns in DB is applied to my schema file and jumbles up all my comments and chosen order in file.

It seems that Prisma currently uses DB (Postgres in my case) as source of truth for column order, and not the file. Postgres does not have a way to change column ordering.

However, it is not sustainable to use introspection for what it is meant to be, once DB is in production and your schema file evolves in time. 

## Suggested solution
I wish it was configurable, or introspection was tracking the lines/columns such that the location of column definition in my file stays irrespective of column order.

Treat the schema file as `source of truth` for schema, which is why we intend to introspect to update the `source`.

## Alternatives
So far, during dev, I have recreated my DB (on a greenfield project). But once in production, I will have to stop using introspection!

## Additional context
prisma/migrate#345

**Comment by janpio** [comment]
Thanks for opening this issue, I applied the appropriate labels - the `introspection-configuration` one includes quite a few similar or related requests.

One question for clarification: "and jumbles up all my comments" - what exactly do you mean by that? Do the comments not stay with the models and fields they belong to?

**Comment by TLadd** [comment]
Yeah, we've been running into this too where multiple developers working on the same project will have a column or two in a different order and the introspection diff moves things around.

**Comment by albertoperdomo** [comment]
Relevant feedback on DB-based field sorting: https://github.com/prisma/prisma/issues/2968#issuecomment-692402345

**Comment by albertoperdomo** [comment]
Relevant feedback from @fchu via https://github.com/prisma/prisma/issues/3659#issuecomment-723364901

>* if there is no pre-existing prisma.schema file, do in the order from the database schema
>* if there is a pre-existing prisma.schema file:
>    * honor any existing fields in the most stable way (eg keep order, intuit renames, etc)
>    * new fields gets added at the end
>
>The thinking here is: if a user is keeping the order in the file, they probably want it. If they want the default db order, they can always get it by temporarily renaming their prisma.schema file.

**Comment by arcticmatt** [comment]
Is this still being worked on? @albertoperdomo @janpio

**Comment by janpio** [comment]
It has never been worked on, as it is a feature request in our GitHub repository. It captures an idea for a feature and now collects reactions, feedback and uses cases that we use to prioritize the things we choose to work on.

**Comment by g0dzillaa** [comment]
@janpio yeah, this is supper annoying as I don't want `created_at` field or any less relevant fields showing up in the first columns.

Interestingly, I found that sometimes when you run `prisma migrate dev`, the order is restored (from the schema file) and when using `prisma db push` (using planetscale), the order always messes up. Note that `prisma migrate dev` does not always keep the order, so it cannot be used as a solution for now (even though the recommended way is to use `db push` for the DB I am using.

**Comment by janpio** [comment]
Can you clarify the connection here @g0dzillaa? The issue is about Introspection and `db pull`, and the Prisma schema file being changed to reflect the order of the database columns.

**Comment by g0dzillaa** [comment]
Hey @janpio,

Missed this message. I am using PlanetScale and using `npx prisma db push` to dev branch and then merge it in the PS console. 

Using Prisma Emulation to enforce FK constraints at the ORM level.

```
datasource db {
  provider     = "mysql"
  url          = env("DATABASE_URL")
  relationMode = "prisma"
}
```


 I am unsure if there is a need for `npx prisma db pull` as I can see all the fields in the dev db but with order changed (sometimes).
 
 I am following: https://www.prisma.io/docs/guides/database/planetscale

**Comment by janpio** [comment]
That was my point: This issue is about Introspection and `db pull` changing the order of the columns in the Prisma schema file based on the order of the columns in the database. I think you are concerned about the other direction (database schema not having the columns being _created_ in the same order as the Prisma schema file defines).

**Comment by g0dzillaa** [comment]
yes, you are right. Ideally, I want to keep the order managed in the schema. The tables are not altered outside of the schema file and I think the tables should respect the column orders as defined in the schema file.

**Comment by amfriedman** [comment]
I come from the PHP world and the phinx migration tool allows control over the column order. For large databases, not being able to see the most important columns first, and inconsistent column ordering conventions across multiple tables is painful to develop with.

**Comment by AlexanderArvidsson** [comment]
This issue is quite old, but is it possible to get a configuration option to sort the fields?

For example; Django has no guarantee of the order the columns are added to Postgres which means every time we pull the DB schema it re-orders the schema fields, making diffs hard to overview changes.
Our code-generator also uses the Prisma schema field order, but since those are changing inconsistently between developers (due to Django not guaranteeing ordering), every commit with a pull causes hundreds of files to change.
